core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksRequest.java

```diff
@@ -63,6 +66,16 @@ public ListTasksRequest setWaitForCompletion(boolean waitForCompletion) {
         return this;
     }
 
+    /**
+     * Returns true if this request tries to lookup a single task
+     */
+    public boolean isLookupTaskRequest() {
+        return false == getTaskId().equals(TaskId.EMPTY_TASK_ID)
+                && getParentTaskId().equals(TaskId.EMPTY_TASK_ID)
+                && isEmpty(this.getNodesIds())
+                && isEmpty(getActions());
+    }
+
```

Comment:
It feels like the single task mode significantly diverged from the list task mode in terms of parameters that we support, functionality that is provided and code that is getting executed in one mode vs another. Handling both in a single action becomes more and more hackish. It might be time to extract the single task mode into a separate `TransportGetTaskAction` with its own request and response. What do you think?